### DISCUSSION TITLE
Creating New Directories While Downloading

### DISCUSSION THREAD

**Discussion by jdduprey** [sr320/course-fish546-2021#25] [Q&A] [answered]
While downloading the sequence and protein I got some "no such directory exists" errors cause I made a mistake when creating my project directory structure. Is there a way to create a new directory automatically when downloading the file `!curl <source file>  > ../<some new directory>`. Or is there a good reason to avoid doing this?

**Comment by jdduprey** [comment]
Some additional questions that came up for me:
* Is there a way to use tab completion in jupyter?
* Could we cover the importance of the step where we append the version number to the blast database?
* Is there a way to clearly see how the blast process is progressing/run it in verbose mode?
* Does zipping/unzipping files affect the checksums?
* When is the appropriate time to create/edit the .gitignore file? I couldn't get my github desktop app to not attempt to upload the large sequence files until I deleted the files themselves from the local repository.

**Comment by kubu4** [comment, marked as answer]
Just documenting discussions from class today:

>  Is there a way to create a new directory automatically when downloading the file

Possibly (I'm not familiar with most of the `curl` commands), but it's probably best practice to establish a directory structure before downloading the files.

> Is there a way to use tab completion in jupyter?

It might take some time to function after starting a notebook. However, you can force it to happen more quickly by pasting the following line (just the highlighted code part, not the bullet point) in a cell at the top of your notebook and running that cell:

- `%config Completer.use_jedi = False`

> Could we cover the importance of the step where we append the version number to the blast database?

The "version" info is very similar to documenting when a BLAST database file was downloaded. Potentially needed for reproducibility.

> Is there a way to clearly see how the blast process is progressing/run it in verbose mode?

Looking at CPU activity (for Macs, the Activity Monitor is useful), frequently checking the line count of the output file (`wc -l outfmt6.tab`), monitoring the progress of the output file (`tail -f outfmt6.tab`), and/or checking the last time the output file was modified (`ls -l outfmt6.tab`).

> Does zipping/unzipping files affect the checksums?

The compressed "version" of a file is considered a different file and will have a different checksum than the uncompressed, original file.

> When is the appropriate time to create/edit the .gitignore file?

The `.gitignore` file needs to be edited _before_ you add a file you might want (need?) to ignore. Hopefully next week's readings on `git` and GitHub will address this.
